### PR TITLE
[#63] Setup Kover and remove Jacoco

### LINKED ISSUE #63: [Chore] Try setup KOver instead of Jacoco for report
## Why

IC project needs UI testing and after trying out some UI cases, I found out currently implemented Jacoco report isn't compatible with UI tests. The tests are failing since the activity doesn't get started and crashing the app. I found a workaround for the issue at [this issue comment](https://github.com/android/android-test/issues/1199#issuecomment-1003197518). But we need that `testCoverageEnabled true` for Jacoco setup. So I will try to implement [Kover](https://github.com/Kotlin/kotlinx-kover) since it doesn't need `testCoverageEnabled` option and also supported by [danger-shroud](https://github.com/livefront/danger-shroud)

## Who Benefits?

If it works then everyone in the android chapter can use this approach for compose projects

### PR DESCRIPTION
**Pull request by kaungkhantsoe**
Closes #63 

## What happened 👀

IC project needs UI testing and after trying out some UI cases, I found out currently implemented Jacoco report isn't compatible with UI tests. The tests are failing since the activity doesn't get started and crashing the app. I found a workaround for the issue at https://github.com/android/android-test/issues/1199#issuecomment-1003197518. But we need that `testCoverageEnabled true` for Jacoco setup. So I will try to implement [Kover](https://github.com/Kotlin/kotlinx-kover) since it doesn't need `testCoverageEnabled` option and also supported by [danger-shroud](https://github.com/livefront/danger-shroud)

## Insight 📝

- Remove Jacoco from the project
- Setup KOver
- Update danger to use KOver report

## Proof Of Work 📹

Please check the result at [this link](https://github.com/kaungkhantsoe/NimbleSurveyJetpackCompose/runs/7781858694?check_suite_focus=true)

### CODE REVIEW

**Review by AVI5HEK — APPROVED** [review]
Kover seems really promising. Let's give it a shot. 🚀

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/splash/SplashScreen.kt [hunk 1]
```diff
@@ -1,3 +1,4 @@
+@file:Suppress("UnusedPrivateMember")
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/splash/SplashScreen.kt]: We shouldn't need this annotation here 🤔

**Comment by ryan-conway** [comment]
Code LGTM, but why are we not merging into `develop` instead? 🤔

**Comment by kaungkhantsoe** [comment]
> Code LGTM, but why are we not merging into `develop` instead? 🤔

At the time of push, `Jacoco branch` hasn't been merged into `develop` yet. So I had to branch from `feature branch`. This will be merged into `develop` as soon as I merged the `base branch`.

**Comment by ryan-conway** [comment]
@kaungkhantsoe Would have made more sense to branch off from `chore/14-setup-jacoco-and-shroud` if it wasn't merged yet since the branches are related 😉

**Review by ryan-conway — APPROVED** [review]

**Comment by kaungkhantsoe** [comment]
> @kaungkhantsoe Would have made more sense to branch off from `chore/14-setup-jacoco-and-shroud` if it wasn't merged yet since the branches are related 😉

@ryan-conway Yes, but I had to continue doing `integrate ticket` and also setup CI for `UI test` of `Splash screen`. In order to do the UI test automation, I need both `KOver` and `UI ticket` of splash screen. That's why I had to branch off from `Splash UI branch`.

**Pull request merged**